core/src/main/scala/cromwell/util/DatabaseUtil.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.util
+
+import java.sql.SQLTransientException
+
+import akka.actor.ActorSystem
+import cromwell.core.retry.{Retry, SimpleExponentialBackoff}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+object DatabaseUtil {
+  private def isTransient(throwable: Throwable): Boolean = throwable match {
+    case _: SQLTransientException => true
+    case _ => false
+  }
+
+  def withRetry[A](f: => Future[A])(implicit actorSystem: ActorSystem): Future[A] = {
```

Comment:
I don't think this should be a thunk, the Future can't be re-run in the `withRetry`.